File: concepts/src/main/java/javaexample/chaining/FutureDummy.java
package javaexample.chaining;

import com.twitter.util.Future;

public class FutureDummy {
    public static Future<Integer> doA(){
        return Future.value(1);
    }
    public static Future<Integer> doB(Integer a){
        return Future.value(a+1);
    }
    public static Future<Integer> doC(Integer b){
        return Future.value(b+1);
    }
    public static Future<Integer> doD(Integer c){
        return Future.value(c+1);
    }
}

File: concepts/src/main/scala/scalaexample/chaining/FutureDummy.scala
package scalaexample.chaining

import com.twitter.util.Future

object FutureDummy {
  def doA() = Future(1)
  def doB(a: Int) = Future(a+1)
  def doC(b: Int) = Future(b+1)
  def doD(c: Int) = Future(c+1)
}
